californium-core/src/test/java/org/eclipse/californium/core/test/lockstep/ObserveClientSideTest.java

```diff
@@ -1327,4 +1327,52 @@ public void testNotificationIsNotHandledAsProactiveCancelResponse() throws Excep
 		assertEquals(mid, cancelResponse.getMID());
 		assertResponseContainsExpectedPayload(cancelResponse, cancelPayload);
 	}
+
+	/**
+	 * Ensure that response with unexpected observe option is not ignored
```

Comment:
Can you add the hint, that this is only intended to ensure compatibility with a broken/none RFC-7252 compliant client?

Reply:
I agree this test is a bit strange and a bit out of spec, I mainly write it to ensure that the issue reported in #688 was fixed.

I can understand, if you think this should not be integrated in the californium test base.

Reply:
From my side there is no objection to have this test included to ensure, your already analysed issue is still fixed. A added hint would ensure, that no one takes this as compliant example.

Reply:
Ok I will add this !

Reply:
> Ok I will add this !

Any due day? I would prefer to have such a comment added before merging